### DISCUSSION TITLE
Autocomplete or custom rendering

### DISCUSSION THREAD

**Discussion by Zipexpo** [pmndrs/leva#317] [Q&A] [answered]
Any idea how to make a autocomplete input ( the same as https://mui.com/components/autocomplete/#main-content )
or how to make a custom render?
![image](https://user-images.githubusercontent.com/18387748/156959309-bf39cb33-9d6d-4c2e-bccb-77a1cf207959.png)

If you have any documents on how to make the custom plugin please share them to me. Thank you for reading.

**Comment by dbismut** [comment, marked as answer]
@Zipexpo Leva supports plugin. Please have a look at https://github.com/pmndrs/leva/tree/main/packages/plugin-bezier or any other packages as a start and feel free to ask specific question in this thread.

**Reply by Zipexpo** [reply]
SO basically, I can use https://github.com/pmndrs/leva/tree/main/packages/plugin-bezier as an example and start building the plugin?
I will try and come back if I have any questions. Thank you so much

p/s: it will be awsome if we can have document about create plugin somewhere

**Reply by dbismut** [reply]
Yes. Or any other plugin in the packages repository!

**Reply by Zipexpo** [reply]
Thank! It worked!
![image](https://user-images.githubusercontent.com/18387748/157825038-22cc4397-1db2-48ff-8cfb-b807751911a7.png)

Although, I still don't get what is the difference of **sanitize** and **normalize**

**Reply by dbismut** [reply]
@Zipexpo Have a look at the comments here

https://github.com/pmndrs/leva/blob/f5b86132e61f09e516a26aec036140a76a72bfbd/packages/leva/src/types/public.ts#L282-L305
